agent/lib/kontena/helpers/observable.rb

```diff
@@ -0,0 +1,167 @@
+module Kontena
+  # An Actor that has some value
+  # The value does not yet exist when initialized, it is nil
+  # Once the value is first updated, then other Actors will be able to observe it
+  # When the value later updated, other Actors will also observe those changes
+  module Observable
```

Comment:
Module namespace does not match filesystem.. I think we could put this directly under `lib/kontena/`.